### PR TITLE
19 slu status

### PR DESCRIPTION
**Pull request by JuriBog**

### CODE REVIEW

File: slu-status/src/slu-status/slu-status.service.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { Status } from './model/Status';
+
+@Injectable()
+export class SluStatusService {
+	private readonly logger = new Logger(SluStatusService.name);
+
+	constructor(@InjectModel(SluStatus.name) private sluStatusModel: Model<SluStatus>) {}
+
+	async saveSluStatus(sluStatus: SluStatusDto): Promise<SluStatus> {
+		try {
```
Comment (mastrogiovanni) [inline slu-status/src/slu-status/slu-status.service.ts:15]: Please simplify the method: the body should be simply:
```
return await new this.sluStatusModel(sluStatus).save();
```

File: slu-status/src/slu-status/slu-status.service.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+			const savedSluStatus = (await new this.sluStatusModel(sluStatus).save()).toObject();
+			delete savedSluStatus._id;
+			return savedSluStatus;
+		} catch (ex: any) {
+			this.logger.error(ex.message);
+			throw new BadRequestException(ex.message);
+		}
+	}
+	async updateSluStatus(id: string, status: Status): Promise<SluStatus> {
+		try {
```
Comment (mastrogiovanni) [inline slu-status/src/slu-status/slu-status.service.ts:25]: Please simplify the method: the body should be simply:
```
return await this.sluStatusModel.findByIdAndUpdate(id, { status: status }, { new: true });
```

File: slu-status/src/slu-status/slu-status.service.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+			const updatedSluStatus = (await this.sluStatusModel.findOneAndUpdate({ id: id }, { status: status }, { new: true })).toObject();
+			delete updatedSluStatus._id;
+			return updatedSluStatus;
+		} catch (ex: any) {
+			this.logger.error(ex.message);
+			throw new BadRequestException(ex);
+		}
+	}
+	async getSluStatus(id: string): Promise<string> {
+		try {
```
Comment (mastrogiovanni) [inline slu-status/src/slu-status/slu-status.service.ts:35]: Please simplify the method: the body should be simply:
```
const slu = await this.sluStatusModel.findById(id).lean();
return slu?.status;
```

**Review by mastrogiovanni — CHANGES_REQUESTED** [review]

File: slu-status/package.json [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    "start:debug": "nest start --debug --watch",
+    "start:prod": "node dist/main",
+    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "test:cov": "jest --coverage",
+    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
+    "test:e2e": "DATABASE_URL=mongodb://127.0.0.1:6000/ jest --config ./test/jest-e2e.json --detectOpenHandles --forceExit"
+  },
+  "dependencies": {
```
[the remark this answers is not in the document]
Comment (JuriBog) [reply slu-status/package.json:23]: it is at line 28:   "@nestjs/mongoose": "^9.0.2"

File: slu-status/src/slu-status/slu-status.service.ts (hunk 2 above)
Comment (JuriBog) [reply slu-status/src/slu-status/slu-status.service.ts:25]: I am using findOneAndUpdate({ id: id }, { status: status }, { new: true, fields: { _id: 0 } }) 
I assume that the user who is requesting this microservice will use the id(did:iota:....)  of the SLU and not the ObjectId of Mongo. Or is the assumption wrong?

File: slu-status/package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (mastrogiovanni) [reply slu-status/package.json:23]: As in the documentation https://docs.nestjs.com/techniques/mongodb you need to have both
@nestjs/mongoose and mongoose
Maybe you already have mongoose installed globally so you didn't notice the error.

[the remark this answers is not in the document]
Comment (JuriBog) [reply slu-status/package.json:23]: Ah ok understand. Is installed now.

**Review by dominic22 — APPROVED** [review]

**Pull request merged**